timetable/utils.py

```diff
@@ -48,13 +48,19 @@ def get_timetable_context(lessons):
 
     # TODO: a cleaner way to pass period str to the template while using
     #       period number as key?
+    alternative_periods = Period.objects.exclude(schedule=1) # Exclude default periods
+    alternative_day_plan = None
+    for day_plan in DayPlan.objects.all():
+        if day_plan.is_today:
+            alternative_day_plan = day_plan
+            break
+
     table = OrderedDict()
     for period in periods:
         table[period] = (period_strs[period], OrderedDict())
-        for dayPlan in DayPlan.objects.all():
-            if dayPlan.is_today and not dayPlan.schedule.is_default:
-                for alt_period in alternative_periods.filter(number=period):
-                    table[period] = table[period] + (alt_period,)
+        if alternative_day_plan is not None:
+            alt_period = alternative_periods.filter(number=period, schedule=alternative_day_plan.schedule)[0]
```

Comment:
Note that the tuple `(number, schedule)` is unique_together [here](https://github.com/zorzalo3/zorza/blob/master/timetable/models.py#L103), so using `filter` method is irrlevant. Additionally, you are wrongly assuming that for every period in default schedule there is equivalent period in alterantive schedule. For example during shortened lessons, the 9th and 10th lessons are cancelled (they are not included in the shortened schedule). You can cross out the periods that are cancelled (when `alternative_periods.get(number=period, schedule=alternative_day_plan.schedule)` throws `DoesNotExist` error). This is why I got IndexError - you tried to access the first element of empty list (`alternative_periods.filter(number=10, schedule=44)` is empty queryset).
![image](https://github.com/zorzalo3/zorza/assets/44808028/3446cb24-510d-4898-8554-11b748476bd7)
**Please test your changes before comitting them.**